meteor/sofie-server/data-access/repositories/test/mongo-rundow-repository.spec.ts

```diff
@@ -0,0 +1,177 @@
+			const segmentRepository: SegmentRepository = mock<SegmentRepository>()
+			const rundownId: string = 'someRundownId'
+			const activeRundown: Rundown = createRundown({ rundownId: rundownId, isRundownActive: true })
+			await testDatabase.populateDatabaseWithRundowns([activeRundown])
+			const db: Db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertRundown(anything())).thenReturn(activeRundown)
+			when(segmentRepository.getSegments(anything())).thenResolve([])
+			const testee = await createCommonTestee({
+				segmentRepository: segmentRepository,
+				mongoConverter: mongoConverter,
+			})
+			await testee.deleteRundown(rundownId)
+
+			expect(await db.collection(COLLECTION_NAME).countDocuments()).toBe(0)
+		})
+
+		it('deletes inactive rundown successfully', async () => {
+			const mongoConverter = mock(MongoEntityConverter)
+			const segmentRepository: SegmentRepository = mock<SegmentRepository>()
+			const rundownId: string = 'someRundownId'
+			const inactiveRundown: Rundown = createRundown({ rundownId: rundownId, isRundownActive: false })
+			await testDatabase.populateDatabaseWithRundowns([inactiveRundown])
+			const db: Db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertRundown(anything())).thenReturn(inactiveRundown)
+			when(segmentRepository.getSegments(anything())).thenResolve([])
+			const testee = await createCommonTestee({
+				segmentRepository: segmentRepository,
+				mongoConverter: mongoConverter,
+			})
+			await testee.deleteRundown(rundownId)
+
+			expect(await db.collection(COLLECTION_NAME).countDocuments()).toBe(0)
+		})
+
+		// eslint-disable-next-line jest/expect-expect
+		it('calls deletion of segments', async () => {
+			const mongoConverter = mock(MongoEntityConverter)
+			const segmentRepository: SegmentRepository = mock<SegmentRepository>()
+			const rundownId: string = 'someRundownId'
+			const rundown: Rundown = createRundown({ rundownId: rundownId })
+			await testDatabase.populateDatabaseWithRundowns([rundown])
+
+			when(mongoConverter.convertRundown(anything())).thenReturn(rundown)
+			when(segmentRepository.getSegments(anything())).thenResolve([])
+			const testee = await createCommonTestee({
+				segmentRepository: segmentRepository,
+				mongoConverter: mongoConverter,
+			})
+			await testee.deleteRundown(rundownId)
+
+			verify(segmentRepository.deleteRundownSegments(anyString())).once()
+		})
+
+		it('does not delete, when nonexistent rundownId is given', async () => {
+			const nonExistingId: string = 'nonExistingId'
+			const rundown: Rundown = createRundown({})
+			await testDatabase.populateDatabaseWithRundowns([rundown])
+			const db: Db = testDatabase.getDatabase()
+
+			const testee = await createCommonTestee({})
+
+			expect.assertions(2)
+			try {
+				await testee.deleteRundown(nonExistingId)
+			} catch (error) {
+				// eslint-disable-next-line jest/no-conditional-expect
+				expect(error).toBeInstanceOf(NotFoundException)
```

Comment:
If some other error is thrown we will still hit the catch, and most likely pass the check that nothing was deleted.
It that case, i'd call it a false positiv, as the test passed, but not because it threw the exception that we expected.

Reply:
This one can be disregard completly, thanks to your other suggestion that has been applied across all the relevant tests